test/upgrading/upgrader.spec.ts

```diff
@@ -499,6 +499,13 @@ describe('libp2p.upgrader', () => {
       libp2p.components.getUpgrader().upgradeOutbound(outbound),
       remoteLibp2p.components.getUpgrader().upgradeInbound(inbound)
     ])
+    await new Promise<void>((resolve, reject) => {
```

Comment:
Please use `pEvent` instead of the hand-rolled promise here, it's more concise and better reflects the intention behind the code.